kotlinx-coroutines-debug/src/CoroutinesBlockHoundIntegration.kt

```diff
@@ -1,16 +1,53 @@
 @file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
+
 package kotlinx.coroutines.debug
 
-import reactor.blockhound.BlockHound
 import kotlinx.coroutines.scheduling.*
+import reactor.blockhound.*
 import reactor.blockhound.integration.*
 
 @Suppress("UNUSED")
-public class CoroutinesBlockHoundIntegration: BlockHoundIntegration {
+public class CoroutinesBlockHoundIntegration : BlockHoundIntegration {
 
-    override fun applyTo(builder: BlockHound.Builder) {
-        builder.addDynamicThreadPredicate { isSchedulerWorker(it) }
-        builder.nonBlockingThreadPredicate { p -> p.or { mayNotBlock(it) } }
+    override fun applyTo(builder: BlockHound.Builder): Unit = with(builder) {
+        // These classes use a lock internally, but should be safe to use.
+        for (method in listOf(
```

Comment:
Currently, it's a bit hard to read and get through. Could you please split it to a few methods?
Like
```
allowBlockingCallsInsideChannelImpl()
allowBlockingCallsInsideReflection()
... etc. ...
```